daemons/attrd/attrd_commands.c

```diff
@@ -628,6 +628,11 @@ attrd_peer_message(crm_node_t *peer, xmlNode *xml)
             /* Synchronize if there is an attribute held only by own node that Writer does not have. */
             attrd_current_only_attribute_update(peer, xml);
         }
+
+    } else if (pcmk__str_eq(op, PCMK__ATTRD_CMD_FLUSH, pcmk__str_casei)) {
```

Comment:
I'm not entirely sure why we use `pcmk__str_casei` so often when values are both set by and checked against defined constant. I think it was a safety measure during an automated change (Coccinelle IIRC) when overhauling the string libraries.

I used it here for consistency, but let me know if there's a preference to start omitting it when it isn't necessary.

Reply:
pcmk__str_eq() was a more flexible replacement for safe_str_eq(), which was always case-insensitive. That was actually a big reason why we replaced it, because it was too convenient to use safe_str_eq() everywhere, even when case-sensitive would be better. We used pcmk__str_casei wherever safe_str_eq() was used previously, to preserve backward compatibility and avoid surprises, but at some point we'll need to examine all the calls closely and decide where/when we can make them case-sensitive.

Here, we could drop it since (as you mentioned) it's always been a constant, and we don't support users using the IPC APIs directly (only via the Pacemaker libraries or command-line tools). But it's not important.

Reply:
As an example of the headaches involved, the many comparisons to CIB XML tag names should always be case-sensitive, but since we allow users to run CIBs with schema validation disabled, it would break backward compatibility to do that now. We'll have to wait to change those until a major-version schema upgrade with an XSL transform converting all tag names to lowercase, even though the chances of anyone actually mis-casing tag names with validation disabled is near zero.

We could change internal-only stuff like IPC API handling anytime (where the casing has always been well-defined of course).

Reply:
To further digress, we could get away with fixing some things (like the CIB tag names) anyway, if we consider the existing usages as bugs rather than features. But I'm not in a hurry to prioritize a significant project just to shave a few CPU cycles.

Reply:
:/

How you keep up with all of the fun and obscure ways in which we could break compatibility is one of life's many mysteries.